miasm2/core/graph.py

```diff
@@ -162,42 +148,113 @@ def dot(self):
         return out
 
 
+    def reachable_sons(self, head):
+        """Compute every nodes reachable from node @head"""
 
-    def compute_dominators(self):
-        """Compute dominators of the graph"""
-        nodes = self.nodes()
-        heads = self.heads()
+        todo = set([head])
+        reachable = set()
+        while todo:
+            node = todo.pop()
+            if node in reachable:
+                continue
+            reachable.add(node)
+            yield node
+            for succ in self.successors_iter(node):
+                todo.add(succ)
+
+    def reachable_parents(self, leaf):
+        """Compute every parents of node @leaf"""
 
+        todo = set([leaf])
+        reachable = set()
+        while todo:
+            node = todo.pop()
+            if node in reachable:
+                continue
+            reachable.add(node)
+            yield node
+            for pred in self.predecessors_iter(node):
+                todo.add(pred)
+
+    def compute_dominators(self, head):
+        """Compute dominators of the graph"""
+        nodes = set(self.reachable_sons(head))
         dominators = defaultdict(set)
```

Comment:
Where did you use the fact that dominators is a `defaultdict`?
You can use `set` for copy in line 184, and avoid using again a `dict()` at the end of the function. In addition, `defaultdict` can easily be prone to error.